src/main/java/fr/moribus/imageonmap/commands/maptool/UpdateCommand.java

```diff
@@ -0,0 +1,137 @@
+ */
+
+package fr.moribus.imageonmap.commands.maptool;
+
+import fr.moribus.imageonmap.Permissions;
+import fr.moribus.imageonmap.commands.IoMCommand;
+import fr.moribus.imageonmap.image.ImageRendererExecutor;
+import fr.moribus.imageonmap.image.ImageUtils;
+import fr.moribus.imageonmap.map.ImageMap;
+import fr.moribus.imageonmap.map.MapManager;
+import fr.zcraft.zlib.components.commands.CommandException;
+import fr.zcraft.zlib.components.commands.CommandInfo;
+import fr.zcraft.zlib.components.i18n.I;
+import fr.zcraft.zlib.components.worker.WorkerCallback;
+import fr.zcraft.zlib.tools.PluginLogger;
+import fr.zcraft.zlib.tools.text.ActionBar;
+import fr.zcraft.zlib.tools.text.MessageSender;
+import org.bukkit.ChatColor;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.HashMap;
+
+@CommandInfo (name =  "update", usageParameters = "<new url> [stretched|covered] \"<map name to update>\"")
+public class UpdateCommand extends IoMCommand
+{
+    @Override
+    protected void run() throws CommandException
+    {
+        final Player player = playerSender();
+        ImageUtils.ScalingType scaling;
+
+        URL url;
+
+        if(args.length < 1) throwInvalidArgument(I.t("You must give an URL and a map name to update."));
+        if(args.length < 2) throwInvalidArgument(I.t("You must give a map name to update."));
+
+        switch(args[1]) {
+
+            case "stretched":
+                scaling = ImageUtils.ScalingType.STRETCHED;
+                break;
+            case "covered":
+                scaling = ImageUtils.ScalingType.COVERED;
+                break;
+            default:
+                scaling = ImageUtils.ScalingType.CONTAINED;
+                break;
+        }
+        ImageMap map;
+        PluginLogger.info("getting mapfromargs");
+        if(scaling.equals(ImageUtils.ScalingType.CONTAINED))
+            map=getMapFromArgs(player,1);
+        else
+            map=getMapFromArgs(player,2);
+        PluginLogger.info("getted mapfromargs");
+        try
+        {
+            url = new URL(args[0]);
+            MapManager.load();
+
+            Integer[] size={1,1};
+            if(map.getType()== ImageMap.Type.POSTER)
+                size=map.getSize( new HashMap<String, Object>(),map.getUserUUID(),map.getId());
+            int width=size[0],height=size[1];
+            try {
+                ActionBar.sendPermanentMessage(player, ChatColor.DARK_GREEN + I.t("Updating..."));
+                ImageRendererExecutor.update(url, scaling, player.getUniqueId(), map, width, height, new WorkerCallback<ImageMap>() {
+                    @Override
+                    public void finished(ImageMap result) {
+                        ActionBar.removeMessage(player);
+                        MessageSender.sendActionBarMessage(player, ChatColor.DARK_GREEN + I.t("Updating finished!"));
```

Comment:
Avis perso, tu en fais ce que tu veux

```suggestion
                        MessageSender.sendActionBarMessage(player, ChatColor.DARK_GREEN + I.t("The map was updated using the new image!"));
```

Reply:
C'est vrai que "Updating finished" ça fait plus penser à une MaJ du plugin et pas une MaJ de la carte.